actions/update-chart-dependency/main.ts

```diff
@@ -0,0 +1,418 @@
+  octokit: ReturnType<typeof getOctokit>,
+  owner: string,
+  repo: string,
+  branchName: string,
+  dependency: string,
+  version: string,
+  fileUpdates: { path: string; content: string }[]
+): Promise<void> {
+  for (const { path: filePath, content } of fileUpdates) {
+    const fileSha = await getFileSha(octokit, owner, repo, filePath, branchName);
+    await octokit.rest.repos.createOrUpdateFileContents({
+      owner,
+      repo,
+      path: filePath,
+      message: `deps: update '${dependency}' to version ${version} in ${filePath}`,
+      content: Buffer.from(content).toString('base64'),
+      branch: branchName,
+      sha: fileSha,
+      committer: {
+        name: 'github-actions[bot]',
+        email: 'github-actions[bot]@users.noreply.github.com',
+      },
+      author: {
+        name: 'github-actions[bot]',
+        email: 'github-actions[bot]@users.noreply.github.com',
+      },
+    });
+  }
+}
+
+/**
+ * @description Create a pull request using GitHub API
+ * @param octokit - github client
+ * @param owner - repo owner
+ * @param repo - repo name
+ * @param branchName - branch name
+ * @param chartsUpdated - list of chart directories updated
+ * @param dependencyName - dependency name
+ * @param newVersion - new chart version
+ * @param baseBranch - base branch name
+ */
+async function createPullRequest(
+  octokit: ReturnType<typeof getOctokit>,
+  owner: string,
+  repo: string,
+  branchName: string,
+  chartsUpdated: string[],
+  dependencyName: string,
+  newVersion: string,
+  baseBranch: string
+): Promise<void> {
+  await octokit.rest.pulls.create({
+    owner,
+    repo,
+    title: `${PR_TITLE_PREFIX}${dependencyName}`,
+    head: branchName,
+    base: baseBranch,
+    body: `Update Helm chart dependency '${dependencyName}' to version ${newVersion} in charts: ${chartsUpdated.join(', ')}.`,
+  });
+}
+
+/**
+ * @description Main action runner
+ * @returns Promise<void>
+ */
+export async function run(): Promise<void> {
+  try {
+    const { serviceName, version, githubToken, chartPrefix, branch } = getInputs();
+
+    const missingInputs = !serviceName || !version || !githubToken;
+    if (missingInputs) {
+      setFailed('Missing required inputs: service-name, version, github-token');
+      return;
+    }
+
+    // Initialize GitHub API client and repo context
+    const octokit = getOctokit(githubToken);
+    const { owner, repo } = githubContext.repo;
+
+    // Determine workspace directory (GitHub Actions or local)
+    const workspace = process.env.GITHUB_WORKSPACE ?? process.cwd();
+
+    // Collect all chart/helmfile files to process, filtered by prefix if provided
+    const chartFilesWithDirs = getChartFilesWithDirs(workspace, chartPrefix);
+
+    if (chartFilesWithDirs.length === 0) {
+      info(`No charts required updating for dependency '${serviceName}'. No PR will be opened.`);
+      return;
+    }
+
+    const fileUpdates: { path: string; content: string }[] = [];
```

Comment:
you use this type multiple times, extract it to one instead of inline

Reply:
Added.